### ISSUE TITLE
Wait for nickserv identification to complete before joining channels

### ISSUE THREAD

**Issue by tylerstillwater** [psychs/limechat#150]
There are some channels that require you to be registered before allowing you to join. The auto-join feature attempts to join immediately upon connecting, causing many channels to deny the join attempt.

**Comment by strcmp** [comment]
yes but it appears as if the SASL username is always assumed to be your nickname which is not always the case. It would be cool if a custom username could be provided, or if the 'username' field was used for this.
